react-app-drawer-component/my-app/src/Overlay.tsx

```diff
@@ -0,0 +1,13 @@
+import './Overlay.css';
+type Props = {
+  isOpen: boolean;
+  onMain: () => void;
+};
+
+export function Overlay({ isOpen, onMain }: Props) {
+  return isOpen ? (
+    <div className="semi-transparent" onClick={onMain}></div>
+  ) : (
+    <div className="transparent" onClick={onMain}></div>
+  );
```

Comment:
The only difference between these two branches is the class name. It would be best to compute the class name in the body of the function and then have just one JSX that uses that class name.